src/Ocelot/Configuration/Creator/HttpVersionPolicyCreator.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace Ocelot.Configuration.Creator;
+
+/// <summary>
+/// Default implementation of the <see cref="IVersionPolicyCreator"/> interface.
+/// </summary>
+public class HttpVersionPolicyCreator : IVersionPolicyCreator
+{
+    /// <summary>
+    /// Creates a <see cref="HttpVersionPolicy"/> by a string.
+    /// </summary>
+    /// <param name="downstreamVersionPolicy">The string representation of the version policy.</param>
+    /// <returns>An <see cref="HttpVersionPolicy"/> enumeration value.</returns>
+    public HttpVersionPolicy Create(string downstreamVersionPolicy) => downstreamVersionPolicy switch
+    {
+        VersionPolicies.RequestVersionExact => HttpVersionPolicy.RequestVersionExact,
+        VersionPolicies.RequestVersionOrHigher => HttpVersionPolicy.RequestVersionOrHigher,
+        VersionPolicies.RequestVersionOrLower => HttpVersionPolicy.RequestVersionOrLower,
+        _ => HttpVersionPolicy.RequestVersionOrLower,
```

Comment:
The creator will return the `RequestVersionOrLower` as default value!
But the author, Ibnu, wanted another value, `RequestVersionOrHigher`.

So, here is the quote from description:

>   changes:
>   - added `DefaultRequestVersion = downstreamRoute.DownstreamHttpVersion,`  and `DefaultVersionPolicy = HttpVersionPolicy.RequestVersionOrHigher,` in `HttpClient` instantiation.
> 
> Then all those components above can talk to each other in HTTP/2.
> 
> And I think `DefaultVersionPolicy = HttpVersionPolicy.RequestVersionOrHigher` could be `DefaultVersionPolicy downstreamRoute.DownstreamHttpVersionPolicy`.
> 

I don't see any code with `DefaultVersionPolicy` name or method. What I see now is that in the switch construction, default section returns:
```C#
_ => HttpVersionPolicy.RequestVersionOrLower,
```
If `string downstreamVersionPolicy` argument of the `Create` is empty, absent property in route opts, or the value is not from the list, then the policy will be `RequestVersionOrLower` by default.
**Is it right to return `RequestVersionOrLower` as default one currently?**
Or did we totally miss this requirement?

Reply:
@raman-m we took the default behavior, as documented by Microsoft here: https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpversionpolicy?view=net-8.0

Reply:
Understood! Not an issue...